### PR TITLE
keybase,kbfs,keybase-gui: 5.3.1 -> 5.4.2

### PR DESCRIPTION
**Pull request by Avaq**
###### Motivation for this change

Keybase version 5.4.x offers improved conflict resolution for KBFS. Running this version, I was able to resolve a long standing file conflict previously unresolvable due to various issues.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

----

I have tested these changes by manually running `keybase service`, `kbfsfuse`, and `keybase-gui` and using the GUI to resolve some KBFS file conflicts I had, and send some chat messages. In doing so, I have verified that all the important bits are working.

Binaries untested:

- `kbnm` in `keybase`
- `git-remote-keybase` in `kbfs`
- `redirector` in `kbfs`

----

EDIT: I am running this in my system now via an overlay, and have used other features, including the encrypted git remotes. No problems encountered.

### CODE REVIEW

**Comment by Avaq** [comment]
I wasn't sure whether to add myself to package maintainers or not, but I'd be happy to help maintaining these packages, so I did. Please let me know if I misunderstood. :P

File: pkgs/tools/security/keybase/gui.nix [hunk 1]
```diff
@@ -4,16 +4,16 @@
...
 
 stdenv.mkDerivation rec {
   pname = "keybase-gui";
-  version = "5.3.1"; # Find latest version from https://prerelease.keybase.io/deb/dists/stable/main/binary-amd64/Packages
+  version = "5.4.2"; # Find latest version from https://prerelease.keybase.io/deb/dists/stable/main/binary-amd64/Packages
 
   src = fetchurl {
     url = "https://s3.amazonaws.com/prerelease.keybase.io/linux_binaries/deb/keybase_${version + "-" + versionSuffix}_amd64.deb";
-    sha256 = "00k1xg27arbqpa836c55fhkxjvjbhjgkf6jqzprjzz76zksfdcx4";
+    sha256 = "fac7adf20e43e1a23d757bf7ac7a9011ee7a15388ddbc1faa735959473d5331a";
```
Comment (Avaq) [inline pkgs/tools/security/keybase/gui.nix:16]: Wait a minute - I don't think this is the right hash. How is this even working?

Comment (bennofs) [reply pkgs/tools/security/keybase/gui.nix:16]: The second one seems to be hex encoded, while the first is base32 I believe

Comment (Avaq) [reply pkgs/tools/security/keybase/gui.nix]: I pushed an amendment updating the encoding to base64, and I only now realize you wrote base **32**. How odd.

Comment (Avaq) [reply pkgs/tools/security/keybase/gui.nix]: Okay base32 now :)

**Comment by Avaq** [comment]
@bennofs I switched to base32 for the hash. I think this PR is ready. I've been using this version of Keybase for 5 days using an overlay:

```nix
{ config, ... }:
{
  nixpkgs.overlays = [
    (self: super: let
      url = https://github.com/NixOS/nixpkgs/archive/3d067a7539860593cf60785e09d270c3c66c721f.tar.gz;
      unstable = import (fetchTarball url) { config = config.nixpkgs.config; };
    in {
      keybase = unstable.keybase;
      keybase-gui = unstable.keybase-gui;
      kbfs = unstable.kbfs;
    })
  ];
}
```

**Comment by bennofs** [comment]
Thanks! I'll do a quick smoketest and then merge tomorrow

**Comment by Avaq** [comment]
I just noticed this special treatment for Darwin:

https://github.com/NixOS/nixpkgs/blob/2c1d9afe876df2e49882f21213a1b2acae30f69a/pkgs/top-level/all-packages.nix#L4488-L4495

Headers don't seem to have changed between Keybase versions, so I think we're good:

- old: https://github.com/keybase/client/blob/68bb8c893c5214040d86ea36f2f86fbb7fac8d39/go/chat/attachments/preview_darwin.go#L7
- 5.4.2: https://github.com/keybase/client/blob/7b0bbf1e3c9485eb2e8471752984bac7b043a1e3/go/chat/attachments/preview_darwin.go#L7

But I don't *really* understand what's going on there, and I can't test on Darwin. It might be worth testing on Darwin if someone can. :man_shrugging:

**Comment by Avaq** [comment]
Can anyone just request Graham of Borg to build the package? Should I try it? :o

**Comment by bennofs** [comment]
@GrahamcOfBorg build keybase kbfs keybase-gui

**Pull request merged**